### PR TITLE
added ReleaseOnTouchEnd to PressableButton & Unittests for release behavior

### PR DESCRIPTION
**Pull request by thalbern**
## Overview
- added ReleaseOnTouchEnd to allow sending release event if button touch ended without passing release plane
- on retracting of the button the release event will now fire as soon as we're passing the release distance
- default behavior of this flag is set to true as this was reported as a bug several times already and i think this should be the default behavior.

## Changes
- Fixes: #4565 && #4266 

## Verification
- added play mode test that checks release behavior:
-- normal release
-- release while moving out of the button
-- disabling flag and moving out of the button to check if release isn't triggered in that case
all tests are executed once in normal and once in low frame rate scenario

### CODE REVIEW

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs [hunk 1]
```diff
@@ -51,6 +51,11 @@ public class PressableButton : MonoBehaviour, IMixedRealityTouchHandler
         [FormerlySerializedAs("returnRate")]
         private float returnSpeed = 25.0f;
 
+        [SerializeField]
+        [Tooltip("Button will send the release event on touch end after successful press even if release plane hasn't been passed.")]
+        private bool releaseOnTouchEnd = true;
+        public bool ReleaseOnTouchEnd { get => releaseOnTouchEnd; set => releaseOnTouchEnd = value; }
```
Comment (julenka) [inline Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs]: doc comment so people can see in the api docs as well?

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PressableButtonTests.cs [hunk 2]
```diff
@@ -327,6 +327,91 @@ public IEnumerator ScaleLocalDistances()
...
+            Vector3 rightOfButtonPress = new Vector3(1.0f, 0, 0.01f);
+            Vector3 inButtonOnRelease = new Vector3(0, 0, 0.005f);
+
+            // test scenarios in normal and low framerate
+            int[] stepVariations = { 30, 2 };
+            for (int i = 0; i < stepVariations.Length; ++i)
+            {
+                int numSteps = stepVariations[i];
+                // test release
+                yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSimulationService);
```
Comment (julenka) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/PressableButtonTests.cs]: Suggestion: you can now use TestHand class Luis wrote to make hand movement easier.

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PressableButtonTests.cs [hunk 3]
```diff
@@ -327,6 +327,91 @@ public IEnumerator ScaleLocalDistances()
...
+                yield return PlayModeTestUtilities.MoveHandFromTo(startHand, inButtonOnPress, numSteps, ArticulatedHandPose.GestureId.Open, Handedness.Right, inputSimulationService);
+                yield return PlayModeTestUtilities.MoveHandFromTo(inButtonOnPress, inButtonOnRelease, numSteps, ArticulatedHandPose.GestureId.Open, Handedness.Right, inputSimulationService);
+                yield return PlayModeTestUtilities.HideHand(Handedness.Right, inputSimulationService);
+
+                //yield return new WaitForFixedUpdate();
```
Comment (julenka) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/PressableButtonTests.cs]: remove comments, here and below.

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PressableButtonTests.cs [hunk 4]
```diff
@@ -327,6 +327,91 @@ public IEnumerator ScaleLocalDistances()
...
+            buttonComponent.ButtonReleased.AddListener(() =>
+            {
+                buttonReleased = true;
+            });
+
+            var inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+            Vector3 startHand = new Vector3(0, 0, 0);
+            Vector3 inButtonOnPress = new Vector3(0, 0, 0.01f);
+            Vector3 rightOfButtonPress = new Vector3(1.0f, 0, 0.01f);
+            Vector3 inButtonOnRelease = new Vector3(0, 0, 0.005f);
```
Comment (googlan) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/PressableButtonTests.cs]: Nit: Maybe leave a comment why these numbers are these numbers? (e.g. they are default distances for pressable button)?

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs (hunk 1 above)
Comment (thalbern) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs]: added comments to the other public fields for pressable button as well

**Comment by thalbern** [comment]
@julenka tested on device - works fine also with bad framerate (~20fps with debug / il2cpp debugging)

**Pull request merged**